internal/fileutils/fileutils.go

```diff
@@ -103,3 +105,28 @@ func Lrename(oldPath, newPath string) error {
 
 	return os.Rename(oldPath, newPath)
 }
+
+// LockDir creates a lock file in the specified directory and acquires an exclusive lock on it.
+// It blocks until the lock is available and returns an unlock function to release the lock.
+func LockDir(dir string) (func() error, error) {
+	lockPath := dir + "/.lock"
```

Comment:
```suggestion
	lockPath := filepath.Join(dir, ".lock")
```

Do we also want to crate the dir if does not exist or we are fine with just failing there?

Reply:
> 	lockPath := filepath.Join(dir, ".lock")

done

> Do we also want to crate the dir if does not exist or we are fine with just failing there?

mmh, I feel like creating the directory should be the responsibility of the caller. not a strong opinion though.